crates/pathfinder/src/state/sync.rs

```diff
@@ -41,13 +41,29 @@ impl Default for State {
     }
 }
 
-pub async fn sync(
+pub async fn sync<Transport, SequencerClient, F1, F2, L1Sync, L2Sync>(
     storage: Storage,
-    transport: Web3<Http>,
+    transport: Web3<Transport>,
     chain: Chain,
-    sequencer: sequencer::Client,
+    sequencer: SequencerClient,
     state: Arc<State>,
-) -> anyhow::Result<()> {
+    l1_sync: L1Sync,
+    l2_sync: L2Sync,
+) -> anyhow::Result<()>
+where
+    Transport: web3::Transport,
+    SequencerClient: sequencer::ClientApi + Clone + Send + Sync + 'static,
+    F1: Future<Output = anyhow::Result<()>> + Send + 'static,
+    F2: Future<Output = anyhow::Result<()>> + Send + 'static,
```

Comment:
Do these have to be generic?

Reply:
I'm afraid so: it's `Send + 'static` imposed by `tokio::spawn` that forces these.